plugins/restful/RestfulDataProviderDbQuery.php

```diff
@@ -341,37 +355,35 @@ protected function replace($id) {
    * {@inheritdoc}
    */
   public function update($id, $full_replace = FALSE) {
-    $query = db_update($this->getTableName());
-    foreach ($this->getIdColumn() as $index => $column) {
-      $query->condition($column, current($this->getColumnFromIds(array($id), $index)));
-    }
-
     // Build the update array.
     $request = $this->getRequest();
     static::cleanRequest($request);
     $public_fields = $this->getPublicFields();
-    $fields = array();
+    $id_columns = $this->getIdColumn();
+
+    $record = array();
     foreach ($public_fields as $public_property => $info) {
       // If this is the primary field, skip.
       if ($this->isPrimaryField($info['property'])) {
         continue;
       }
-      // Check if the public property is set in the payload.
-      if (!isset($request[$public_property])) {
-        if ($full_replace) {
```

Comment:
Are we losing this functionality?

`$full_replace` is what separates a `PUT` request from a `PATCH` request.